src/js/popup.js

```diff
@@ -93,7 +93,14 @@
 
   function getCurrentUSDValue() {
     currencyService.getCurrentUSDValue()
-      .then(onGetCurrentUSDValueSuccess);
+      .then(onGetCurrentUSDValueSuccess)
+      .fail(onGetCurrentUSDValueFail);
+  }
+
+  function onGetCurrentUSDValueFail() {
+    currentUSDValueElement
+      .text('Failed to fetch value')
```

Comment:
I'd write *Failed to get current USD value*. If it happens in the first time some user opens the popup, he doesn't even know that value relates to the current USD value. So, this way, we are more specific and, consequently, clear.